### DISCUSSION TITLE
Jupyterlab Xeus-Python on Jetbot 0.4.3

### DISCUSSION THREAD

**Discussion by Hoehlenbaer** [NVIDIA-AI-IOT/jetbot#421] [Q&A] [answered]
Hey all!

I'm using a kit from "Yahboom", which originally came with the jetbot 0.4.0 image. To get a dashboard with voila running, I updated to Jupyterlab 3 and decided to also update to the latest jetbot release (I like the docker concept!). 

My build has got some more motors and other HW, so my configuration doesn't match (Errno 121 or motors running in the wrong direction, but everything's fine with the original image). 

While trying to debug the Jupyter notebooks and the python files, I realized that a visual debugger might be helpful - here's where the trouble started...

In a nutshell: Is there anyone able to get Xeus-Python running on a jetbot? A conda installation in a docker container isn't working, the pypi-installation can't find matching distributions and a source build fails due to missing dependencies...

So I'm still struggling to find the source of my Errno 121.

**Comment by tokk-nv** [comment, marked as answer]
Hi hoehlenbaer,

I need to say it's generally better to stick with the image that came with the 3rd party kit, as that is designed to take care of the particular hardware configuration. And I'm not sure what you are trying to achieve with Voila and Xeus-Python.
That being said, and you would ultimately need to check with Yahboom on what modifications they made to the jetbot card image to support their customized hardware, it looks like they have a different I2C address for PCA9685.
http://www.yahboom.net/xiazai/Jetbot%20car/Download/Jetbot%20Hardware%20Manual.pdf

So rather than working on Jupyter notebooks, I think it is easy to diagnose on a terminal using `i2cdetect` to check if you can communicate to all the I2C devices on the hardware.
You can check this article on Jetsonhacks.
https://www.jetsonhacks.com/2019/07/22/jetson-nano-using-i2c/

**Reply by Hoehlenbaer** [reply]
Hi tokk-nv,
Thanks for pointing to these pages. Actually after understanding the differences in the hardware configuration, it was pretty easy to solve. The robot is running now with the latest jetbot image...
